src/Android/Xamarin.Android/Samples/Network Analysis/NavigateRouteRerouting/NavigateRouteRerouting.cs

```diff
@@ -0,0 +1,345 @@
+using Esri.ArcGISRuntime.UI;
+using Esri.ArcGISRuntime.UI.Controls;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Threading.Tasks;
+
+namespace ArcGISRuntimeXamarin.Samples.NavigateRouteRerouting
+{
+    [Activity(ConfigurationChanges = Android.Content.PM.ConfigChanges.Orientation | Android.Content.PM.ConfigChanges.ScreenSize)]
+    [ArcGISRuntime.Samples.Shared.Attributes.Sample(
+        "Navigate route with rerouting",
+        "Network Analysis",
+        "Navigate between two points and dynamically recalculate an alternate route when the original route is unavailable.",
+        "")]
+    [ArcGISRuntime.Samples.Shared.Attributes.OfflineData("567e14f3420d40c5a206e5c0284cf8fc", "91e7e6be79cc4d2f8416eff867674c1e")]
+    public class NavigateRouteRerouting : Activity, TextToSpeech.IOnInitListener
+    {
+        // Hold references to the UI controls.
+        private MapView _myMapView;
+        private Button _navigateButton;
+        private Button _recenterButton;
+        private TextView _status;
+
+        // Variables for tracking the navigation route.
+        private RouteTracker _tracker;
+        private RouteResult _routeResult;
+        private Route _route;
+        private RouteTask _routeTask;
+        private RouteParameters _routeParams;
+
+        // List of driving directions for the route.
+        private IReadOnlyList<DirectionManeuver> _directionsList;
+
+        // Speech synthesizer to play voice guidance audio.
+        private TextToSpeech _textToSpeech;
+
+        // String to hold route status for the UI.
+        private string _statusMessage;
+
+        // Graphics to show progress along the route.
+        private Graphic _routeAheadGraphic;
+        private Graphic _routeTraveledGraphic;
+
+        // San Diego Convention Center.
+        private readonly MapPoint _conventionCenter = new MapPoint(-117.160386727, 32.706608, SpatialReferences.Wgs84);
+
+        // RH Fleet Aerospace Museum.
+        private readonly MapPoint _aerospaceMuseum = new MapPoint(-117.146678, 32.730463, SpatialReferences.Wgs84);
+
+        // GPX file that contains simulated location data for driving route.
+        private readonly string _gpxPath = DataManager.GetDataFolder("91e7e6be79cc4d2f8416eff867674c1e", "navigate_a_route_detour.gpx");
+
+        // Path to the network geodatabase for San Diego.
+        private readonly string _networkGeodatabasePath = DataManager.GetDataFolder("567e14f3420d40c5a206e5c0284cf8fc", "sandiego.geodatabase");
+
+        protected override void OnCreate(Bundle bundle)
+        {
+            base.OnCreate(bundle);
+
+            Title = "Navigate route with rerouting";
+
+            CreateLayout();
+            Initialize();
+        }
+
+        private async void Initialize()
+        {
+            try
+            {
+                // Create the map view.
+                _myMapView.Map = new Map(Basemap.CreateNavigationVector());
```

Comment:
I recommend using a raster basemap on Android and Forms to prevent issues with some network configurations.